### DISCUSSION TITLE
Search for all relevant matches not length based hits

### DISCUSSION THREAD

**Discussion by RK-BAKU** [milvus-io/milvus#20913] [Q&A and General discussion]
Actually, I don't know if this is considered a normal search behavior but when I search for some part of the text I get only relevant matches of similar length. How to avoid this and get all relevant results not only limited to the length?
My search database is a comments of varying length. For example:

`Perhaps one of the most insightful things Todd said that I have never really thought about before, is how video games (unlike other forms of media) give the consumer a sense of pride. Films, TV and Books will rarely if ever give you the same feeling of accomplishment that even the most simple of achievements gives the player whilst playing video games, truly insightful!`

if I search for "Films, TV and Books will rarely if ever give" in most cases it doesn't include any matches where the search string is somewhere in the middle of the text. Besides, it will first (and only?) return matches where the text has similar length as my search input.

Is it possible to do something similar to "full-text" search like we have in relational databases or something as a "like" queries?

**Comment by yhmo** [comment]
What you said is like text search, not vector search. Text search is to find/compare the characters or words in the full sentence.
Vector search is different, what you need to do is:
1. split the sentence into some short sentences
2. convert the short sentences to vectors, to binary vectors(by rule) or float vectors(by trained model)
3. insert the vectors into a vector database
4. given a sentence, convert the sentence to a vector, search this vector in vector database

**Reply by RK-BAKU** [reply]
> What you said is like text search, not vector search. Text search is to find/compare the characters or words in the full sentence. Vector search is different, what you need to do is:
> 
> 1. split the sentence into some short sentences
> 2. convert the short sentences to vectors, to binary vectors(by rule) or float vectors(by trained model)
> 3. insert the vectors into a vector database
> 4. given a sentence, convert the sentence to a vector, search this vector in vector database

Hey @yhmo ,
Thank you for answers!
Kindly advise on length of such chunks? Does it matter how to split long sentence to shorts at all ?
For instance, what would be a relevant chunks as for example given above in order to cover whole comment?
